salt/tests/unit/modules/test_metalk8s_kubernetes_utils.py

```diff
@@ -0,0 +1,221 @@
+                ), patch("kubernetes.client.VersionApi", api_instance_mock):
+            if raises:
+                self.assertRaisesRegexp(
+                    CommandExecutionError,
+                    result,
+                    metalk8s_kubernetes_utils.get_version_info
+                )
+            else:
+                self.assertEqual(
+                    metalk8s_kubernetes_utils.get_version_info(**kwargs),
+                    result
+                )
+            api_instance_mock.return_value.get_code.assert_called()
+
+    @parameterized.expand([
+        ({"go_version": "my_go_version"}, True),
+        (None, False, True, True),
+    ])
+    def test_ping(
+        self,
+        version_info,
+        result,
+        error=False,
+        raises=False,
+    ):
+        """
+        Tests the return of `ping` function
+        """
+        version_info_mock = MagicMock()
+        if error:
+            version_info_mock.side_effect = CommandExecutionError(
+                'Failed to get version info'
+            )
+        version_info_mock.return_value.get_version_info.return_value = \
+            version_info
+        with patch(
+            "metalk8s_kubernetes_utils.get_version_info", version_info_mock
+        ):
+            if raises:
+                self.assertEqual(
+                    result,
+                    metalk8s_kubernetes_utils.ping()
+                )
+            else:
+                self.assertTrue(
+                    metalk8s_kubernetes_utils.ping(),
+                )
+
+    @parameterized.expand([
+        param.explicit(kwargs=test_case)
+        for test_case in YAML_TESTS_CASES["read_and_render_yaml_file"]
+    ])
+    def test_read_and_render_yaml_file(
+        self,
+        source,
+        result,
+        context=None,
+        template=None,
+        raises=False,
+        **kwargs
+    ):
+        """
+        Tests the return of `read_and_render_yaml_file` function
+        """
+        cache_file_mock = MagicMock()
+        open_file_mock = mock_open(read_data=yaml.dump(source))
+
+        if source is None:
+            cache_file_mock.side_effect = CommandExecutionError(
+                'Source file my-source-file not found'
+            )
+        else:
+            cache_file_mock.return_value = source
+
+        patch_dict = {
+            'cp.cache_file': cache_file_mock
+        }
+        with patch.dict(metalk8s_kubernetes_utils.__salt__, patch_dict), \
+                patch("salt.utils.files.fopen", open_file_mock):
+            if raises:
+                self.assertRaisesRegexp(
+                    CommandExecutionError,
+                    result,
+                    metalk8s_kubernetes_utils.read_and_render_yaml_file,
+                    source="my_source",
+                    template=template,
+                    **kwargs
+                )
+            else:
+                self.assertDictContainsSubset(
```

Comment:
To me you should use `assertEqual` here we want to have the full expected dict as output not only a part of it